src/tactic/show_term.lean

```diff
@@ -0,0 +1,41 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import tactic.core
+import tactic.tauto
+
+namespace tactic
+
+/-- Run a tactic, returning the goals as they were before running the tactic. -/
+meta def capture_terms {α : Type} (t : tactic α) : tactic (list expr × α) :=
+do
+  gs ← get_goals,
+  a ← t,
+  return (gs, a)
+
+/-- Run a tactic, and then return the pretty printed original goal. -/
+meta def pp_term {α : Type} (t : tactic α) : tactic string :=
+do
+  g :: _ ← get_goals,
+  t,
+  to_string <$> pp g
+
+namespace interactive
+
+/--
+`show_term { tac }` runs the tactic `tac`,
+and then prints the term that was constructed.
+
+This is useful for understanding what tactics are doing,
+and how metavariables are handled.
+
+As an example, if the goal is `ℕ × ℕ`, `show_term { split, exact 0 }` will
+print `(0, ?m_1)`, indicating that the original goal has been partially filled in.
```

Comment:
When I first saw the PR I assumed that `show_term { tac }` would execute `tac` 1) with a new meta-variable and 2) discard the results.  I think it would help me if you could write here that after you run `show_term`, there is then one goal with target `ℕ`.

Reply:
Okay, I've updated the doc-string.

My idea was this was an "in-place" inspector for tactic proofs you're trying to diagnose. (I actually wrote this because I'd managed to break `tauto` while working on #2222, and needed to be able to see what it was doing when it was working properly.)